modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/resource/OpenAPIResourceImpl.java

```diff
@@ -518,20 +518,14 @@ private String _getBasePath(
 			return UriInfoUtil.getBasePath(uriInfo);
 		}
 
-		String scheme = Http.HTTP;
+		UriBuilder uriBuilder = UriInfoUtil.getBaseUriBuilder(uriInfo);
+
+		uriBuilder.host(_portal.getForwardedHost(httpServletRequest));
 
 		if (_portal.isSecure(httpServletRequest)) {
-			scheme = Http.HTTPS;
+			uriBuilder.scheme(Http.HTTPS);
```

Comment:
I assume that scheme is `HTTP` by default right?

Reply:
I assume we should not change, so it is managed by the UriInfoUtil in this method https://github.com/liferay/liferay-portal/blob/7a541f63ee4b774288dbc2673142961c493a059d/modules/apps/portal-vulcan/portal-vulcan-api/src/main/java/com/liferay/portal/vulcan/util/UriInfoUtil.java#L44